common/src/main/java/com/wynntils/models/stats/StatCalculator.java

```diff
@@ -0,0 +1,197 @@
+        int last;
+        if (showBestValueLastAlways || isGood) {
+            first = valueRange.low();
+            last = valueRange.high();
+        } else {
+            // Emulate Wynncraft behavior by showing the value closest to zero first
+            first = valueRange.high();
+            last = valueRange.low();
+        }
+        // We store "inverted" stats (spell costs) as positive numbers internally,
+        // but need to display them as negative numbers
+        if (statType.showAsInverted()) {
+            first = -first;
+            last = -last;
+        }
+        displayRange = Pair.of(first, last);
+        return displayRange;
+    }
+
+    public static float getPercentage(StatActualValue actualValue, StatPossibleValues possibleValues) {
+        int min = possibleValues.range().low();
+        int max = possibleValues.range().high();
+
+        float percentage = MathUtils.inverseLerp(min, max, actualValue.value()) * 100;
+        return percentage;
+    }
+
+    public static double getPerfectChance(StatPossibleValues possibleValues) {
+        // FIXME: This is the chance of getting a *** (3 star) roll, not the chance of
+        // getting the maximum possible value. But for now, keep old behavior.
+        double perfectChance = 1 / (possibleValues.baseValue() > 0 ? 101d : 61d) * 100;
+        return perfectChance;
+    }
+
+    public static double getDecreaseChance(StatActualValue actualValue, StatPossibleValues possibleValues) {
+        // If we are in any of these situations, our stat has "invalid" values. Maybe it is old?
+        // If so, rerolling will have a certain outcome of 100% or 0% chance of decrease
+        if (actualValue.value() > possibleValues.range().high()) {
+            return 1d;
+        } else if (actualValue.value() < possibleValues.range().low()) {
+            return 0d;
+        }
+
+        // We should not be checking this if it is fixed, but handle that case as well
+        if (possibleValues.range().isFixed()) {
+            return 0d;
+        }
+
+        // This code finds the lowest possible and highest possible rolls that achieve the correct
+        // result (inclusive). Then, it finds the average decrease and increase afterwards
+        RangedValue innerRollRange = actualValue.internalRoll();
+
+        // FIXME: What we probably really want is the percentage of possible internal rolls
+        // that is lower than innerRollRange.low, but do not change this for now.
+        double result;
+        double avg = (innerRollRange.low() + innerRollRange.high()) / 2d;
+        if (innerRollRange.low() > 0) {
+            result = (avg - 30) / 101d;
+        } else {
+            result = (130 - avg) / 61d;
+        }
+        return result * 100;
+    }
+
+    public static double getIncreaseChance(StatActualValue actualValue, StatPossibleValues possibleValues) {
+        // If we are in any of these situations, our stat has "invalid" values. Maybe it is old?
```

Comment:
Comment on decrease chances applies here as well

Reply:
And my reply as well :)